src/main/java/de/ait/secondlife/services/OfferServiceImpl.java

```diff
@@ -327,9 +350,24 @@ private void checkOfferIfIsFree(BigDecimal startPrice, BigDecimal winBin) {
         }
     }
 
+    private void checkUserId(Long id) {
+        if (id == null) throw new IdIsNullException();
+        if (!userService.checkEntityExistsById(id)) throw new UserNotFoundException(id);
+    }
+
     @Override
     public boolean checkEntityExistsById(Long id) {
         if (id == null) throw new IdIsNullException();
         return offerRepository.existsByIdAndIsActiveTrue(id);
     }
+
+    private void checkUserCredentials(Long id) {
+        try {
+            User user = userService.getAuthenticatedUser();
+            if (!user.getId().equals(id))
+                if (!user.getAuthorities().contains(Role.ROLE_ADMIN))
+                    throw new NoRightToChangeException("User  is not authorised for this operation");
```

Comment:
why do you use this type of exception here? no modification operations are expected